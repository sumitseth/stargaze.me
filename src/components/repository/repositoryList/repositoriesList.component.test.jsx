import React from 'react';
import { render, getByTestId } from '@testing-library/react';
import renderer from 'react-test-renderer';
import RepositoriesList from './repositoriesList.component';
import { repositoriesListTestSelectors } from './respositoryList.testSelectors';

const repository = {
  id: '1',
  name: 'stargaze.me',
  url: 'http://stargaze.me',
  description: 'star',
  viewerHasStarred: true,
};

it('renders without crashing', () => {
  const { container } = render(
    <RepositoriesList repositories={{ nodes: [repository] }} />
  );
  const list = getByTestId(container, repositoriesListTestSelectors.list);
  expect(list.childElementCount).toBe(1);
});

it('matches snapshot', () => {
  const tree = renderer
    .create(<RepositoriesList repositories={{ nodes: [repository] }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
