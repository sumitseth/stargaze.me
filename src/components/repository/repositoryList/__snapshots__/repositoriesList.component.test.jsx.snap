// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<ul
  className="sc-htoDjs htrlDF"
  data-testid="repositories-list"
>
  <div
    className="sc-bdVaJa sc-gzVnrw beCVhZ"
  >
    <img
      alt="viewerHasStarred"
      onClick={[Function]}
      src={
        Object {
          "0": "s",
          "1": "t",
          "10": "e",
          "11": ".",
          "12": "p",
          "13": "n",
          "14": "g",
          "2": "a",
          "3": "r",
          "4": "-",
          "5": "a",
          "6": "c",
          "7": "t",
          "8": "i",
          "9": "v",
          "default": "star-active.png",
        }
      }
      style={
        Object {
          "cursor": "pointer",
          "verticalAlign": "middle",
        }
      }
    />
    <h2
      data-testid="repository-name"
    >
      stargaze.me
    </h2>
    <a
      data-testid="repository-url"
      href="http://stargaze.me"
      rel="noopener noreferrer"
      target="_blank"
    >
      http://stargaze.me
    </a>
    <p
      data-testid="repository-description"
    >
      star
    </p>
  </div>
</ul>
`;
